core/src/main/scala/org/http4s/UrlForm.scala

```diff
@@ -28,16 +34,17 @@ object UrlForm {
   val empty: UrlForm = new UrlForm(Map.empty)
 
   def apply(values: Map[String, Seq[String]]): UrlForm =
-    // value "" -> Seq() is just noise and it is not maintain during encoding round trip
-    if(values.get("").fold(false)(_.isEmpty)) new UrlForm(values - "")
-    else new UrlForm(values)
+    // value key -> Seq() is just noise and it is not maintain during encoding round trip
+    new UrlForm(values.filter(_._2.nonEmpty))
```

Comment:
I think this is breaking the [RFC 5849 example](https://github.com/http4s/http4s/blob/ec7c928183e5fe2bdc308f565e3ee1c09f561ae7/client/src/test/scala/org/http4s/client/oauth1/OAuthTest.scala#L68): the values can be empty so long as the key is not also empty.